runtime-sdk/src/modules/consensus/mod.rs

```diff
@@ -0,0 +1,279 @@
+};
+
+use crate::{
+    context::{Context, DispatchContext},
+    module,
+    module::Module as _,
+    modules,
+    types::{address::Address, message::MessageEventHookInvocation, token},
+};
+
+#[cfg(test)]
+mod test;
+
+/// Unique module name.
+const MODULE_NAME: &str = "consensus";
+
+/// Parameters for the consensus module.
+#[derive(Clone, Debug, Serialize, Deserialize)]
+#[serde(deny_unknown_fields)]
+pub struct Parameters {
+    pub consensus_denomination: token::Denomination,
+}
+
+impl Default for Parameters {
+    fn default() -> Self {
+        Self {
+            consensus_denomination: token::Denomination::from_str("TEST").unwrap(),
+        }
+    }
+}
+
+impl module::Parameters for Parameters {
+    type Error = ();
+}
+/// Events emitted by the consensus module (none so far).
+#[derive(Debug, Serialize, Deserialize, oasis_runtime_sdk_macros::Event)]
+#[serde(untagged)]
+pub enum Event {}
+
+/// Genesis state for the consensus module.
+#[derive(Clone, Debug, Default, Serialize, Deserialize)]
+#[serde(deny_unknown_fields)]
+pub struct Genesis {
+    #[serde(rename = "parameters")]
+    pub parameters: Parameters,
+}
+
+#[derive(Error, Debug, oasis_runtime_sdk_macros::Error)]
+pub enum Error {
+    #[error("invalid argument")]
+    #[sdk_error(code = 1)]
+    InvalidArgument,
+
+    #[error("invalid denomination")]
+    #[sdk_error(code = 2)]
+    InvalidDenomination,
+
+    #[error("internal state: {0}")]
+    #[sdk_error(code = 3)]
+    InternalStateError(#[from] StateError),
+
+    #[error("core: {0}")]
+    #[sdk_error(transparent)]
+    Core(#[from] modules::core::Error),
+}
+
+/// Interface that can be called from other modules.
+pub trait API {
+    /// Transfer an amount from the runtime account.
+    fn transfer<C: Context>(
+        ctx: &mut C,
+        to: Address,
+        amount: &token::BaseUnits,
+        hook: MessageEventHookInvocation,
+    ) -> Result<(), Error>;
+
+    /// Withdraw an amount into the runtime account.
+    fn withdraw<C: Context>(
+        ctx: &mut C,
+        from: Address,
+        amount: &token::BaseUnits,
+        hook: MessageEventHookInvocation,
+    ) -> Result<(), Error>;
+
+    /// Escrow an amount of the runtime account funds.
+    fn escrow<C: Context>(
+        ctx: &mut C,
+        to: Address,
+        amount: &token::BaseUnits,
+        hook: MessageEventHookInvocation,
+    ) -> Result<(), Error>;
+
+    /// Reclaim an amount of runtime staked shares.
+    fn reclaim_escrow<C: Context>(
+        ctx: &mut C,
+        from: Address,
+        amount: &token::BaseUnits,
+        hook: MessageEventHookInvocation,
+    ) -> Result<(), Error>;
+
+    /// Returns consensus token denomination.
+    fn consensus_denomination<C: Context>(ctx: &mut C) -> Result<token::Denomination, Error>;
+
+    /// Query consensus account info.
+    fn account<C: Context>(ctx: &C, addr: Address) -> Result<ConsensusAccount, Error>;
+
+    // TODO:
+    //  - add more staking queries as needed.
```

Comment:
Should we open an issue for these? Or just remove the comment?

Reply:
Just removed the comment since it's obvious where new queries should be added if ever needed .